lib/elixir/lib/enum.ex

```diff
@@ -1575,51 +1613,74 @@ defmodule Enum do
 
-  If multiple elements are considered minimal, the first one that was found
-  is returned.
+  By default, the comparison is done with the `<=` sorter function.
+  If multiple elements are considered minimal, the first one that
+  was found is returned. If you want the last element considered
+  minimal to be returned, the sorter function should not return true
+  for equal elements.
 
-  Calls the provided `empty_fallback` function and returns its value if
-  `enumerable` is empty. The default `empty_fallback` raises `Enum.EmptyError`.
+  If the enumerable is empty, the provided `empty_fallback` is called.
+  The default `empty_fallback` raises `Enum.EmptyError`.
 
   ## Examples
 
       iex> Enum.min([1, 2, 3])
       1
 
-      iex> Enum.min([], fn -> 0 end)
-      0
-
   The fact this function uses Erlang's term ordering means that the comparison
   is structural and not semantic. For example:
 
       iex> Enum.min([~D[2017-03-31], ~D[2017-04-01]])
       ~D[2017-04-01]
 
-  In the example above, `min/1` returned April 1st instead of March 31st
-  because the structural comparison compares the day before the year. This
-  can be addressed by using `min_by/3` and by relying on structures where
-  the most significant digits come first. In this particular case, we can
-  use `Date.to_erl/1` to get a tuple representation with year, month and day
-  fields:
+  In the example above, `min/2` returned April 1st instead of March 31st
+  because the structural comparison compares the day before the year.
+  For this reason, most structs provide a "compare" function, such as
+  `Date.compare/2`, which receives two structs and returns `:lt` (less than),
+  `:eq` (equal), and `:gt` (greather than). If you pass a module as the
+  sorting function, Elixir will automatically use the `compare/2` function
+  of said module:
 
-      iex> Enum.min_by([~D[2017-03-31], ~D[2017-04-01]], &Date.to_erl/1)
+      iex> Enum.min([~D[2017-03-31], ~D[2017-04-01]], Date)
       ~D[2017-03-31]
 
-  For selecting a minimal value out of two consider using `Kernel.min/2`.
+  Finally, if you don't want to raise on empty enumerables, you can pass
+  the empty callback:
+
+      iex> Enum.min([], &<=/2, fn -> 0 end)
+      0
 
   """
-  @spec min(t, (() -> empty_result)) :: element | empty_result when empty_result: any
-  def min(enumerable, empty_fallback \\ fn -> raise Enum.EmptyError end)
-      when is_function(empty_fallback, 0) do
-    aggregate(enumerable, &Kernel.min/2, empty_fallback)
+  @spec min(t, (element, element -> boolean) | module(), (() -> empty_result)) ::
+          element | empty_result
+        when empty_result: any
+  def min(enumerable, sorter \\ &<=/2, empty_fallback \\ fn -> raise Enum.EmptyError end) do
+    {sorter, empty_fallback} = min_sort_fun(sorter, empty_fallback)
+    aggregate(enumerable, sorter, empty_fallback)
   end
 
+  # TODO: Deprecate me on 1.14
+  defp min_sort_fun(empty_fallback, default_empty_callback)
```

Comment:
Same here, and the lines below.

```suggestion
  defp min_sort_fun(empty_fallback, default_empty_fallback)
```